newdle/client/src/components/answer/AnswerGrid.js

```diff
@@ -0,0 +1,703 @@
+        if (p.id === participant.id) {
+          return {...participant, comment, answers};
+        } else {
+          return p;
+        }
+      });
+    } else {
+      participants = [...participants, {...participant, comment, answers}];
+    }
+  } else if (user) {
+    // user has not answered yet, create a new participant object for them
+    participants = [
+      ...participants,
+      {
+        id: -1,
+        name: user.name,
+        avatar_url: user.avatar_url,
+        auth_uid: user.uid,
+        comment,
+        answers,
+      },
+    ];
+  }
+
+  participants = _.sortBy(participants, ['name']);
+
+  if (unknown) {
+    // keep the unknown participant in the first table row
+    participants = [
+      {
+        id: -1,
+        name,
+        comment,
+        answers,
+      },
+      ...participants,
+    ];
+  }
+
+  return (
+    <div className={styles['answer-grid']}>
+      <Table textAlign="center">
+        <TableHeader timeslots={timeslots} interactive={false} isCreator={false} />
+        <Table.Body>
+          {participants.map(p => {
+            const selectable = participant
+              ? participant.id === p.id
+              : user
+              ? user.uid === p.auth_uid
+              : p.id === -1;
+            return (
+              <AnswerRow
+                key={p.id}
+                participant={p}
+                timeslots={timeslots}
+                answers={answers}
+                selectable={selectable}
+                hasBusyTimes={hasBusyTimes}
+                busyTimes={busyTimes}
+                duration={duration}
+                unknown={unknown && selectable}
+              />
+            );
+          })}
+        </Table.Body>
+        <TableFooter participants={participants} timeslots={timeslots} interactive={false} />
+      </Table>
+    </div>
+  );
+}
+
+AnswerGrid.propTypes = {
+  participant: PropTypes.object,
+  user: PropTypes.object,
+  name: PropTypes.string,
+  comment: PropTypes.string,
+  unknown: PropTypes.bool.isRequired,
+  hasBusyTimes: PropTypes.bool,
+  busyTimes: PropTypes.object.isRequired,
+  duration: PropTypes.number.isRequired,
+};
+
+export function ParticipantGrid({finalDate, setFinalDate, isCreator, finalized}) {
+  const timeslots = useSelector(getNewdleTimeslots);
+  const participants = useSelector(getNewdleParticipants);
+  const [hoveredColumn, setHoveredColumn] = useState(null);
+
+  if (timeslots.length === 0) {
+    return null;
+  }
+
+  return (
+    <div className={styles['participant-grid']}>
+      <Table textAlign="center">
+        <TableHeader
+          timeslots={timeslots}
+          interactive={!finalized}
+          finalDate={finalDate}
+          setFinalDate={setFinalDate}
+          isCreator={isCreator}
+          hoveredColumn={hoveredColumn}
+          setHoveredColumn={setHoveredColumn}
+        />
+        <Table.Body>
+          {_.sortBy(participants, ['name']).map(participant => (
```

Comment:
```suggestion
          {_.sortBy(participants, 'name').map(participant => (
```

AFAIK you do not need an array when sorting by a single field